### ISSUE TITLE
TypeError: Attempted to assign to readonly property

### ISSUE THREAD

**Issue by lms11** [Nozbe/WatermelonDB#895]
Hey there,

I'm having a weird (and hard to debug) a problem with syncing. After it finishes pulling my changes from backend, I get this error: `TypeError: Attempted to assign to readonly property`. This happens after the following call at `Database/ActionQueue.js`:
```
      this._queue.push({
        work: work,
        resolve: resolve,
        reject: reject,
        description: description
      });
```

This is the stack trace I'm getting (happy to symbolise if it helps, but the calls with "→" are on the enqueue at ActionQueue):
```
→ push@[native code] http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:158027:26
tryCallTwo@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:6329:9
doResolve@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:6493:25
Promise@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:6352:14
→ enqueue@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:158015:25
http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:179272:57
[native code]
tryCallOne@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:6320:16
http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:6421:27
_callTimer@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:30868:17
_callImmediatesPass@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:30907:17
callImmediates@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:31124:33
__callImmediates@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:5614:35
http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:5400:34
__guard@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:5597:15
flushedQueue@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:5399:21
flushedQueue@[native code]
invokeCallbackAndReturnFlushedQueue@[native code]
```

I've tried pretty much everything:
- Checking my schema
- Checking my adaptors e database instances
- Reinstalling WatermelonDB

Nothing worked.

While debugging WatermelonDB, I had a strong feeling the issue is with the transpilation -- and I'm not sure how to move forward here so any help is appreciated.

Here's how it looks like on Metro's bundle:
```
var ActionQueue = function () {
    function ActionQueue() {
      this._queue = [];
      this._subActionIncoming = false;
    }

    var _proto = ActionQueue.prototype;

    _proto.enqueue = function enqueue(work, description) {
      var _this = this;

      if (this._subActionIncoming) {
        this._subActionIncoming = false;
        return work(this);
      }

      return new Promise(function (resolve, reject) {
        if ('production' !== process.env.NODE_ENV && _this._queue.length) {
          var queue = _this._queue;
          var current = queue[0];

          _common.logger.warn("[WatermelonDB] The action you're trying to perform (".concat(description || 'unnamed', ") can't be performed yet, because there are ").concat(queue.length, " actions in the queue. Current action: ").concat(current.description || 'unnamed', ". Ignore this message if everything is working fine. But if your actions are not running, it's because the current action is stuck. Remember that if you're calling an action from an action, you must use subAction(). See docs for more details."));

          _common.logger.log("[WatermelonDB] Enqueued action:", work);

          _common.logger.log("[WatermelonDB] Running action:", current.work);
        }

        // Here's where it fails
        _this._queue.push({
          work: work,
          resolve: resolve,
          reject: reject,
          description: description
        });

        if (1 === _this._queue.length) {
          _this._executeNext();
        }
      });
    };
...
```

Thoughts? Has anyone seen this before?

Thanks a lot,
Lucas

**Comment by lms11** [comment]
Just checked with example provided in repository, and the bundle generated also produces the same output as above, invalidating my theory around traspilation issues I guess.

**Comment by lms11** [comment]
Alright, so I managed to fix the issue but I still don't know the exact cause (I'd love to learn, if someone knows!).

I'm also using `react-native-navigation` and I was wrapping each component I'd register into a database provider. That would, somehow, create non-writable versions of the action queue. By passing the database as a prop, it was fixed.

(really, if you have **any** hypothesis, let me know!)
